Client/game_sa/TaskAttackSA.cpp

```diff
@@ -46,398 +38,180 @@ CTaskSimpleUseGunSA::CTaskSimpleUseGunSA(CEntity* pTargetEntity, CVector vecTarg
-    DWORD dwThisInterface = (DWORD)this->GetInterface();
-    _asm
-    {
-        mov     ecx, dwThisInterface
-        call    dwFunc
-        mov     bReturn, al
-    }
-    return bReturn;
+    // CTaskSimpleUseGun::GetIsFiring
+    return ((bool(__thiscall*)(void*))FUNC_CTaskSimpleUseGun_GetIsFiring)(GetInterface());
 }
 
 bool CTaskSimpleUseGunSA::GetIsReloading()
 {
-    bool  bReturn;
-    DWORD dwFunc = FUNC_CTaskSimpleUseGun_GetIsReloading;
-    DWORD dwThisInterface = (DWORD)this->GetInterface();
-    _asm
-    {
-        mov     ecx, dwThisInterface
-        call    dwFunc
-        mov     bReturn, al
-    }
-    return bReturn;
+    // CTaskSimpleUseGun::GetIsReloading
+    return ((bool(__thiscall*)(void*))FUNC_CTaskSimpleUseGun_GetIsReloading)(GetInterface());
 }
 
 bool CTaskSimpleUseGunSA::GetSkipAim()
 {
-    bool  bReturn;
-    DWORD dwFunc = FUNC_CTaskSimpleUseGun_GetSkipAim;
-    DWORD dwThisInterface = (DWORD)this->GetInterface();
-    _asm
-    {
-        mov     ecx, dwThisInterface
-        call    dwFunc
-        mov     bReturn, al
-    }
-    return bReturn;
+    // CTaskSimpleUseGun::GetSkipAim
+    return ((bool(__thiscall*)(void*))FUNC_CTaskSimpleUseGun_GetSkipAim)(GetInterface());
 }
 
 bool CTaskSimpleUseGunSA::PlayerPassiveControlGun()
 {
-    bool  bReturn;
-    DWORD dwFunc = FUNC_CTaskSimpleUseGun_PlayerPassiveControlGun;
-    DWORD dwThisInterface = (DWORD)this->GetInterface();
-    _asm
-    {
-        mov     ecx, dwThisInterface
-        call    dwFunc
-        mov     bReturn, al
-    }
-    return bReturn;
+    // CTaskSimpleUseGun::PlayerPassiveControlGun
+    return ((bool(__thiscall*)(void*))FUNC_CTaskSimpleUseGun_PlayerPassiveControlGun)(GetInterface());
 }
 
 void CTaskSimpleUseGunSA::RemoveStanceAnims(CPed* pPed, float f)
 {
-    DWORD dwFunc = FUNC_CTaskSimpleUseGun_RemoveStanceAnims;
-    DWORD dwThisInterface = (DWORD)this->GetInterface();
-    DWORD dwPedInterface = (DWORD)pPed->GetInterface();
-    _asm
-    {
-        mov     ecx, dwThisInterface
-        push    f
-        push    dwPedInterface
-        call    dwFunc
-    }
+    // CTaskSimpleUseGun::RemoveStanceAnims
+    ((void(__thiscall*)(void*, CPedSAInterface*, float))FUNC_CTaskSimpleUseGun_RemoveStanceAnims)(GetInterface(), pPed->GetPedInterface(), f);
 }
 
 bool CTaskSimpleUseGunSA::RequirePistolWhip(CPed* pPed, CEntity* pTargetEntity)
 {
-    bool  bReturn;
-    DWORD dwFunc = FUNC_CTaskSimpleUseGun_ControlGun;
-    DWORD dwPedInterface = (DWORD)pPed->GetInterface();
-    DWORD dwTargetEntity = (pTargetEntity) ? (DWORD)pTargetEntity->GetInterface() : 0;
-    _asm
-    {
-        push    dwTargetEntity
-        push    dwPedInterface
-        call    dwFunc
-        mov     bReturn, al
-    }
-    return bReturn;
+    CPedSAInterface*    pPedInterface = pPed->GetPedInterface();
+    CEntitySAInterface* pTargetEntityInterface = pTargetEntity ? pTargetEntity->GetInterface() : nullptr;
+
+    // actually thiscall
```

Comment:
If it's thiscall, shouldn't we be using thiscall?

Reply:
CTaskSimpleUseGunSA::RequirePistolWhip is a static function and also unused